CodeGeneration/Sources/generate-swiftsyntaxbuilder/ResultBuildersFile.swift

```diff
@@ -0,0 +1,194 @@
+        FunctionDecl(
+          """
+          /// With buildEither(second:), enables support for 'if-else' and 'switch'
+          /// statements by folding conditional results into a single result.
+          public static func buildEither(first component: Component) -> Component {
+            return component
+          }
+          """
+        )
+        
+        FunctionDecl(
+          """
+          /// With buildEither(first:), enables support for 'if-else' and 'switch'
+          /// statements by folding conditional results into a single result.
+          public static func buildEither(second component: Component) -> Component {
+            return component
+          }
+          """
+        )
+                
+        FunctionDecl(
+          """
+          /// Enables support for 'for..in' loops by combining the
+          /// results of all iterations into a single result.
+          public static func buildArray(_ components: [Component]) -> Component {
+            return components.flatMap { $0 }
+          }
+          """
+        )
+        
+        FunctionDecl(
+          """
+          /// If declared, this will be called on the partial result of an 'if'
+          /// #available' block to allow the result builder to erase type
+          /// information.
+          public static func buildLimitedAvailability(_ component: Component) -> Component {
+            return component
+          }
+          """
+        )
+        
+        FunctionDecl(
+          leadingTrivia: [
+            .newlines(1),
+            .docLineComment("/// If declared, this will be called on the partial result from the outermost"),
+            .newlines(1),
+            .docLineComment("/// block statement to produce the final returned result."),
+            .newlines(1)],
+          modifiers: [Token.public, Token.static],
+          identifier: .identifier("buildFinalResult"),
+          signature: FunctionSignature(
+            input: ParameterClause {
+              FunctionParameter(
+                firstName: .wildcard,
+                secondName: .identifier("component"),
+                colon: .colon,
+                type: "Component")
+            },
+            output: "FinalResult")) {
+              if type.isToken {
+                ReturnStmt(
+                  """
+                  return .init(component)
+                  """
+                )
+              } else if node.collectionElementType.hasWithTrailingCommaTrait {
+                VariableDecl(
+                  """
+                  let lastIndex = component.count - 1
+                  """
+                )
+                
+                ReturnStmt(
+                  """
+                  return .init(component.enumerated().map({ index, source in
+                    let element = source.create\(elementType.buildableBaseName)()
+                    return index < lastIndex ? element.ensuringTrailingComma() : element
+                  }))
```

Comment:
If I'm not mistaken, we should be able to use a trailing closure here

```suggestion
                  return .init(component.enumerated().map { index, source in
                    let element = source.create\(elementType.buildableBaseName)()
                    return index < lastIndex ? element.ensuringTrailingComma() : element
                  })
```